source/server/listener_manager_impl.cc

```diff
@@ -1037,6 +1037,17 @@ Network::ListenSocketFactoryPtr ListenerManagerImpl::createListenSocketFactory(
   }
 }
 
+void ListenerManagerImpl::maybeCloseSocketsForListener(ListenerImpl& listener) {
+  if (!listener.udpListenerConfig().has_value() ||
+      listener.udpListenerConfig()->listenerFactory().isTransportConnectionless()) {
+    // Close the listen socket right away to avoid leaving TCP connections in accept queue
+    // already waiting for long timeout. However, connection-oriented UDP listener shouldn't
+    // close the socket because it needs to receive packets on existing connection via the
```

Comment:
it needs to -> they need to 
on existing connection -> for existing connections

Reply:
done